src/Parser.zig

```diff
@@ -2070,8 +2094,8 @@ fn coerceArrayInit(p: *Parser, item: *Result, tok: TokenIndex, target: Type) !bo
         return true; // do not do further coercion
     }
 
-    if (target.specifier == .array) {
-        assert(item.ty.specifier == .array);
+    if (target.is(.array)) {
+        assert(item.ty.is(.array));
```

Comment:
The use of `target.data.array.len` below should be replaced with `arrayLen` and the assert should be left as is since `item` is supposed to be a string literal.